src/Core/Geometry/MeshPrimitives.cpp

```diff
@@ -140,7 +141,7 @@ TriangleMesh makeGeodesicSphere( Scalar radius, uint numSubdiv ) {
 
             const Scalar x = 2.f * sq5_5 * std::cos( theta );
             const Scalar y = 2.f * sq5_5 * std::sin( theta );
-            const Scalar z = j == 0 ? sq5_5 : -sq5_5;
+            const Scalar z = Math::areApproxEqual( Scalar( j ), 0_ra ) ? sq5_5 : -sq5_5;
```

Comment:
I'm confused here, `j` is an `int`, so the code was right, why casting to `Scalar`?

Reply:
I'm really not an adept of over-typing. I propose to limit explicit typing (preferably with _ra) only where it is mandatory. Let the compiler do the job when it is able to, it does generally better than we. I agree, j is an int and integer comparison is exact. And from my point of view, this loop might be conveniently unrolled.

Reply:
There is no loop here.